pkg/app/template_decoder.go

```diff
@@ -0,0 +1,30 @@
+package app
+
+import (
+	"encoding/json"
+
+	"github.com/atlassian/smith"
+
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/watch"
+)
+
+type templateDecoder struct {
+	decoder *json.Decoder
```

Comment:
what about embedding an io.Closer here

type tempalteDecoder struct {
         io.Closer
         decoder *json.Decoder
}

Then you can delete the Close forwarding method below.